modules/billing-account/budgets.tf

```diff
@@ -22,6 +22,10 @@ locals {
       "$notification_channels:${k}" => v.id
     }
   )
+  project_numbers = {
```

Comment:
`local.ctx` will not contain the internally defined project numbers, you have to merge them in first.

Check https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/project-factory/projects.tf#L51 for comparison

Reply:
I also don't think we should change the semantics of the project numbers context to have a `projects/` prefix: the billing module should interpolate that. We can't expect the caller to know if project numbers should be plain numbers (like in all other modules), or prefixed.

I would do something like this in the billing module instead: `try("projects/${local.ctx.project_numbers[v]}", v)`